consensus/scheme/rolldpos/blockproposal.go

```diff
@@ -62,9 +63,9 @@ func (bp *blockProposal) ProposerAddress() string {
 	return bp.block.ProducerAddress()
 }
 
-func (bp *blockProposal) LoadProto(msg *iotextypes.BlockProposal) error {
-	// TODO: pass the correct EVM network ID at time of newConsensus() or from ctx
-	blk, err := block.NewDeserializer(config.EVMNetworkID()).FromBlockProto(msg.Block)
+func (bp *blockProposal) LoadProto(ctx context.Context, msg *iotextypes.BlockProposal) error {
```

Comment:
`func (bp *blockProposal) LoadProto(msg *iotextypes.BlockProposal, uint32 evmId)`

caller to extrace evmId from ctx, and call `LoadProto`